lib/pdk/cli/console.rb

```diff
@@ -0,0 +1,126 @@
+module PDK::CLI
+  @console_cmd = @base_cmd.define_command do
+    name 'console'
+    usage _('console [console_options]')
+    summary _('(Experimental) Start a session of the puppet debugger console.')
+    default_subcommand 'help'
+    description _(<<-EOF
+The pdk console runs a interactive session of the puppet debugger tool to test out snippets of code, run
+language evaluations, datatype prototyping and much more.  A virtual playground for your puppet code!
+For usage details see the puppet debugger docs at https://docs.puppet-debugger.com.
+
+EOF
+                 )
+    PDK::CLI.puppet_version_options(self)
+    PDK::CLI.puppet_dev_option(self)
+    # we have to skip option parsing because it is expected the user
+    # will be passing additional args that are passed to the debugger
+    skip_option_parsing
+
+    # TODO: using -h or --help skips the pdk help and passes to puppet debugger help
+    run do |_opts, args, _cmd|
+      PDK::CLI::Util.ensure_in_module!(
+        message: _('Console can only be run from inside a valid module directory'),
+        log_level: :fatal,
+      )
+      processed_options, processed_args = process_opts(args)
+
+      PDK::CLI::Util.validate_puppet_version_opts(processed_options)
+
+      PDK::CLI::Util.analytics_screen_view('console', args)
+
+      # TODO: figure out if we need to remove default configs set by puppet
```

Comment:
Should TODO still be here?

Reply:
Yes.  I think the args should be merged

Reply:
Yes.  I think the args should be merged

Reply:
Going to fix this in the next iteration. I really wanted to see how people would use it and get some feedback.